packages/react-native/ReactAndroid/src/main/java/com/facebook/react/ReactInstanceManager.java

```diff
@@ -585,6 +585,38 @@ public void onHostPause() {
     moveToBeforeResumeLifecycleState();
   }
 
+  /**
+   * This method should be called from {@link Activity#onUserLeaveHint()}. It notifies all
+   * listening modules that the user is about to leave the activity. The passed Activity is has to
+   * be the current Activity.
+   *
+   * @param activity the activity being backgrounded as a result of user action
+   */
+  @ThreadConfined(UI)
+  public void onUserLeaveHint(@Nullable Activity activity) {
+    if (mRequireActivity) {
+      Assertions.assertCondition(mCurrentActivity != null);
+    }
+
+    if (mCurrentActivity != null) {
+      Assertions.assertCondition(
+        activity == mCurrentActivity,
+        "Called onUserLeaveHint on an activity that is not the current activity, this is incorrect! "
+          + "Current activity: "
+          + mCurrentActivity.getClass().getSimpleName()
+          + " "
+          + "Leaving activity: "
+          + activity.getClass().getSimpleName());
```

Comment:
@behenate Ok so one of our product was crashing on this assert
I haven't had the time to investigate what's going on, but practically we should understand if this assert is actually necessary.

That's the stacktrace:

```
02-01 14:38:52.320  4500  4522 V fb-android-monitor: java.lang.AssertionError
02-01 14:38:52.320  4500  4522 V fb-android-monitor: 	at com.facebook.infer.annotation.Assertions.assertCondition(Assertions.java:82)
02-01 14:38:52.320  4500  4522 V fb-android-monitor: 	at com.facebook.react.ReactInstanceManager.onUserLeaveHint(ReactInstanceManager.java:598)
02-01 14:38:52.320  4500  4522 V fb-android-monitor: 	at com.facebook.react.ReactDelegate.onUserLeaveHint(ReactDelegate.java:107)
02-01 14:38:52.320  4500  4522 V fb-android-monitor: 	at com.facebook.react.ReactActivityDelegate.onUserLeaveHint(ReactActivityDelegate.java:127)
02-01 14:38:52.320  4500  4522 V fb-android-monitor: 	at com.facebook.react.ReactActivity.onUserLeaveHint(ReactActivity.java:111)
02-01 14:38:52.320  4500  4522 V fb-android-monitor: 	at android.app.Activity.performUserLeaving(Activity.java:8189)
02-01 14:38:52.320  4500  4522 V fb-android-monitor: 	at android.app.Instrumentation.callActivityOnUserLeaving(Instrumentation.java:1520)
02-01 14:38:52.320  4500  4522 V fb-android-monitor: 	at android.app.ActivityThread.performUserLeavingActivity(ActivityThread.java:4659)
02-01 14:38:52.320  4500  4522 V fb-android-monitor: 	at android.app.ActivityThread.handlePauseActivity(ActivityThread.java:4643)
02-01 14:38:52.320  4500  4522 V fb-android-monitor: 	at android.app.servertransaction.PauseActivityItem.execute(PauseActivityItem.java:46)
02-01 14:38:52.320  4500  4522 V fb-android-monitor: 	at android.app.servertransaction.TransactionExecutor.executeLifecycleState(TransactionExecutor.java:176)
02-01 14:38:52.320  4500  4522 V fb-android-monitor: 	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:97)
02-01 14:38:52.320  4500  4522 V fb-android-monitor: 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2066)
02-01 14:38:52.320  4500  4522 V fb-android-monitor: 	at android.os.Handler.dispatchMessage(Handler.java:106)
02-01 14:38:52.320  4500  4522 V fb-android-monitor: 	at android.os.Looper.loop(Looper.java:223)
02-01 14:38:52.320  4500  4522 V fb-android-monitor: 	at android.app.ActivityThread.main(ActivityThread.java:7656)
02-01 14:38:52.320  4500  4522 V fb-android-monitor: 	at java.lang.reflect.Method.invoke(Native Method)
02-01 14:38:52.320  4500  4522 V fb-android-monitor: 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:592)
02-01 14:38:52.320  4500  4522 V fb-android-monitor: 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:947)
02-01 14:38:52.320  4500  4522 V fb-android-monitor: 	at Z.init(unknown)
```

Could you resubmit 3b6c522 without this assert?

Reply:
@behenate are you able to follow-up here?

Reply:
@cortinico Sorry for no reply, I'm currently out of office until February 14th, I will follow up when I'm back :)

Reply:
Sure no rush 👍